source/hwIo.py

```diff
@@ -312,6 +312,16 @@ def setFeature(self, report):
 				log.debug("Set feature failed: %s" % ctypes.WinError())
 			raise ctypes.WinError()
 
+	def setOutputReport(self,data):
+		"""
+		Write the given data to the device using HidD_SetOutputReport.
+		This is instead of using the standard WriteFile which may freeze with some USB HID implementations.

```

Comment:
It might be good to mention here that this function is not asynchronous, in which aspect it differs from using writefile. It should therefore be used with caution.

Reply:
You might be right... though I found it hard to find documentation that stated this really did block. My testing seems to show it is always instantainious... Though I'm happy to put the warning there as I guess it is possible.